serde/src/private/ser.rs

```diff
@@ -1356,6 +1356,25 @@ where
     }
 }
 
+pub enum VariantName {
+    String(&'static str),
+    Integer(i64),
+    Boolean(bool),
+}
+
+impl Serialize for VariantName {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            VariantName::String(s) => serializer.serialize_str(s),
+            VariantName::Integer(i) => serializer.serialize_i64(*i),
```

Comment:
I am not thrilled about using i64 as the one true type of integer enum tags. For example if someone writes `serde(rename = 100u16)` it would be weird for that to serialize using i64. I'd like to consider respecting the integer suffix, and for enums where all the integers are unsuffixed, use the smallest type that fits all the given values. We do this for extern enums in https://github.com/dtolnay/cxx and it seems to match what people expect.